src/custom_nodes/object_detection/object_detection.cpp

```diff
@@ -0,0 +1,340 @@
+    // cv::Mat image = nhwc_to_mat(imageTensor);  // here you can support other layouts
+
+    NODE_ASSERT(image.cols == imageWidth, "Mat generation failed");
+    NODE_ASSERT(image.rows == imageHeight, "Mat generation failed");
+
+    uint64_t detectionsCount = detectionTensor->dims[2];
+    uint64_t featuresCount = detectionTensor->dims[3];
+
+    std::vector<cv::Rect> boxes;
+    std::vector<float> confidences;
+
+    for (uint64_t i = 0; i < detectionsCount; i++) {
+        float* detection = (float*)(detectionTensor->data + (i * featuresCount * sizeof(float)));
+        int imageId = static_cast<int>(detection[0]);
+        int labelId = static_cast<int>(detection[1]);
+        float confidence = detection[2];
+        int xMin = static_cast<int>(detection[3] * imageWidth);
+        int yMin = static_cast<int>(detection[4] * imageHeight);
+        int xMax = static_cast<int>(detection[5] * imageWidth);
+        int yMax = static_cast<int>(detection[6] * imageHeight);
+        if (imageId == 0 && confidence >= confidenceThreshold) {
+            if (filterLabelId != -1 && filterLabelId != labelId) {
+                if (debugMode) {
+                    std::cout << "Skipping label ID: " << labelId << std::endl;
+                }
+                continue;
+            }
+            auto box = cv::Rect(cv::Point(xMin, yMin), cv::Point(xMax, yMax));
+            boxes.emplace_back(box);
+            confidences.emplace_back(confidence);
+            if (debugMode) {
+                std::cout << "Detection:\nImageID: " << imageId << "; LabelID:" << labelId << "; Confidence:" << confidence << "; Box:" << box << std::endl;
+            }
+        }
+    }
+
+    NODE_ASSERT(boxes.size() == confidences.size(), "boxes and confidences are not equal length");
+    if (boxes.size() > maxOutputBatch) {
+        boxes.resize(maxOutputBatch);
+        confidences.resize(maxOutputBatch);
+    }
+
+    *outputsCount = 3;
+    *outputs = (struct CustomNodeTensor*)malloc(*outputsCount * sizeof(CustomNodeTensor));
+    NODE_ASSERT((*outputs) != nullptr, "malloc has failed");
+
+    CustomNodeTensor& imagesTensor = (*outputs)[0];
+    imagesTensor.name = OUTPUT_IMAGES_TENSOR_NAME;
+    if (copy_images_into_output(&imagesTensor, boxes, image, targetImageHeight, targetImageWidth, convertToGrayScale)) {
+        free(*outputs);
+        return 1;
+    }
+
+    CustomNodeTensor& coordinatesTensor = (*outputs)[1];
+    coordinatesTensor.name = OUTPUT_COORDINATES_TENSOR_NAME;
+    if (copy_coordinates_into_output(&coordinatesTensor, boxes)) {
+        cleanup(imagesTensor);
+        free(*outputs);
+        return 1;
+    }
+
+    CustomNodeTensor& confidencesTensor = (*outputs)[2];
+    confidencesTensor.name = OUTPUT_CONFIDENCES_TENSOR_NAME;
+    if (copy_confidences_into_output(&confidencesTensor, confidences)) {
+        cleanup(imagesTensor);
+        cleanup(coordinatesTensor);
+        free(*outputs);
+        return 1;
+    }
```

Comment:
I find it a little confusing that  function return evaluated to `true` indicates error. 
I'd rather say we need to handle errors when  `if(!function()) { ... handle errors ... }`
With current implementation I'd say that we do the cleanup when those functions succeed and it's expected.

Reply:
Since API is defined in a way that 0 means OK I wanted to follow the same convention - should we change it?

Reply:
Changed for both - this node and already existing east_ocr node